src/Dispatcher.cpp

```diff
@@ -83,7 +88,7 @@ void DispatcherBase::StartServer(
     if (m_active) return;
     m_active = true;
   }
-  m_server = true;
+  m_networkMode = NT_NET_MODE_SERVER;
```

Comment:
Yeah. I would actually add 2 new states to the enum as well.

NT_NET_MODE_STARTING_SERVER and NT_NET_MODE_STARTING_CLIENT. Set those in StartServer and StartClient, then change them to NT_NET_MODE_SERVER and NT_NET_MODE_CLIENT in `ServerThreadMain` and `ClientThreadMain`